### ISSUE TITLE
Log users IP address for transfers

### ISSUE THREAD

**Issue by cometjo** [Nicotine-Plus/nicotine-plus#806]
I have a massive ebook collection and I've written a grep/sed/uniq script to count the number of uploads merging different versions of the same book. I then post this in my user info. Already I have had someone trying to grab the same book simultaneously from different accounts but the same IP address. Logging the users IP in the transfers log would go someways to prevent vandalism.

I realize this is not very pertinent but I figured I'd put it out there.

**Comment by mathiascode** [comment]
Does anyone else here have any opinions for/against this?

**Comment by woollarding** [comment]
i have had users try do this to me, they claim it will get them what they want faster. It can be annoying when the user is very determined...

**Comment by hboetes** [comment]
I think you correctly recognized a problem, but maybe there is a more elegant solution? This one involves watching the logs.
Maybe it's an idea to automatically ban people who try to connect with multiple accounts whilst using the same IP?

I also noticed people who don't share anything, try to download a specific folder, I ban them, they create a new account and try again later.

I bet there are other variations of this sort of abuse. See also #366

**Comment by cometjo** [comment]
Sorry, I hardly ever check the email that was used to set up this account. I've since abandoned my whole stats scripting. I have had a number of thoughts on expanding the 'leech detector' plugin to address this and other issues such as users with some but not many files shared or users that are not connectable (the plugins/examplars folder has code for something like this). I'm not big on banning but I will ban a person by both username and ip usually only for a week or so. It's when they start queuing up gbs of data all at once that irks me. If somebody wants just a handful of ebooks I don't really care. But I haven't touched python since about 2008 and I was hardly a novice then. I got as far as modifying it to look for less than 1000 files before I realized I was in deeper than the time I felt I had to devote to this.
